pkg/config/polling_manager.go

```diff
@@ -205,7 +186,41 @@ func NewPollingProjectConfigManager(sdkKey string, pollingMangerOptions ...Optio
 	}
 
 	initDatafile := pollingProjectConfigManager.initDatafile
-	pollingProjectConfigManager.SyncConfig(initDatafile) // initial poll
+	if len(initDatafile) == 0 {
+		pollingProjectConfigManager.notificationCenter = registry.GetNotificationCenter(sdkKey)
+		pollingProjectConfigManager.SyncConfig() // initial poll
+	} else {
+		if projectConfig, _ := datafileprojectconfig.NewDatafileProjectConfig(initDatafile); projectConfig != nil {
```

Comment:
lines 193 -198 are repeated in NewAsync, maybe it will be better to create a small help function for that.